proposals/nnnn-remote-control-onRcStatus-notification.md

```diff
@@ -0,0 +1,92 @@
+* Status: **Awaiting review**
+* Impacted Platforms: [Core / iOS / Android / RPC ]
+
+## Introduction
+One simple resource allocation rule is that at any given time only one application controls a RC module. For example, if SDL allocates a climate control module to application ONE, SDL rejects other applications to change the settings of the same climate control module. The module is freed when application ONE exits, disconnects, goes to HMI level NONE or no longer has the right to control the module due to various reasons (for example, a policy update revokes the access right to the module from the app, or SDL re-allocates the module to another app). In the last case, the mobile application does not aware the change.
+
+
+A new notification originated from SDL is nice to have. If SDL allocates a module to an application or it deallocates a module from an application, SDL shall send OnRCStatus notifications to both the mobile application and the HMI. The notification shall contain an array of the RC modules allocated to the application, so that the mobile application can know it gains or loses control of a module before a failure request, and the HMI can update the status on which application controls which modules.
+
+## Motivation
+
+Mobile applications needs to know which remote control modules it can control. HMI needs to show which application controls which module or modules.
+
+SDL core manages the remote control modules. It allocates a module to an application upon a successful `OnButtonPress` or `SetInteriorVehicleData` request to access a `free` module. The application knows whether it can control the module from the result code of the response message. If the result is `SUCCESS`, the application stays in control of the module and can send more control requests. If the result is anything other than `SUCCESS`, the application cannot control the requested module. It shall stop sending more requests to the same module unless it goes to HMI level `FULL`. 
+
+SDL can free the allocated resource for various reasons. In most of the cases, the application knows it no longer has the control of the module. However, in some cases, the application does not know it loses the control of the module. For example, a policy update can revoke the application's access to a module. In another example, when a foreground application tries to access an allocated resource, SDL can de-allocate the module from a background application and re-allocate the module to the foreground application. It happens either automatically (the `RCAccessMode` of RC settings is configured as `AUTO_ALLOW`) or manually with the driver's allowing the control right transfer via a popup (the `RCAccessMode` of RC settings is configured as `ASK_DRIVER`). However, in both examples, the application that previously controls the module does not know the change until it receives a failure response on next control request. The HMI also does not aware the change.
```

Comment:
Please update the last sentence to "The HMI is also not aware of the change."

Reply:
changed, thanks